sdk/program/src/nonce/state/mod.rs

```diff
@@ -74,17 +56,34 @@ impl Versions {
                     // upgrade Uninitialized legacy nonces.
                     State::Uninitialized => None,
                     State::Initialized(ref mut data) => {
-                        data.durable_nonce = DurableNonce::from_blockhash(
-                            &data.blockhash(),
-                            true, // separate_domains
-                        );
+                        data.durable_nonce = DurableNonce::from_blockhash(&data.blockhash());
                         Some(Self::Current(state))
                     }
                 }
             }
             Self::Current(_) => None,
         }
     }
+
+    // Updates authority pubkey on the nonce account.
+    pub fn authorize(self, authority: Pubkey) -> Option<Self> {
```

Comment:
I think this method is doing too much. How about having a `update_state` method that just allows updating the state with a passed `state` argument while still preserving the version variant? I think that would result in less refactoring. The tests using this method could use a test-only authorize function if needed

Reply:
Similar to the other comment, I would rather not to expose any Current vs Legacy logic outside of this module;

Reply:
My suggestion doesn't result in exposing that logic outside of this module. It would be something like this:

https://github.com/solana-labs/solana/blob/52aa5acdf134a0888b6741f4a335c9dcf3911734/sdk/program/src/nonce/state/mod.rs#L68-L75

https://github.com/solana-labs/solana/blob/52aa5acdf134a0888b6741f4a335c9dcf3911734/runtime/src/nonce_keyed_account.rs#L272-L278

Reply:
The downside of
```rust
fn update(self, state: Box<State>) -> Self
```

is that the api by itself is ambiguous why we need the state when `self` already has the state and then which state comes out at the end in `-> Self`? the one in `self` or in the other argument.

I have updated this code to encapsulates `AuthorizeNonceAccount` implementation in the nonce module.

Reply:
I don't think there is any ambiguity of what happens when you pass an argument to a method called "update." But your latest change looks good, happy to go with that.